jitbuilder/control/Jit.cpp

```diff
@@ -229,4 +229,6 @@ internal_shutdownJit()
 
    TR::CodeCacheManager &codeCacheManager = fe->codeCacheManager();
    codeCacheManager.destroy();
+
+   TR::CompilationController::shutdown();
```

Comment:
This was a pretty scary one. We are likely not cleaning up a ton of other resources. We really ought to take a hard look at what a proper shutodwn sequence should look like and implement it at a common level. Currently we have a ton of duplicated logic between JitBuilder and TestCompiler for example.

Reply:
Agree. Opened #4749 .